drivers/hmis/spec/models/hmis/form/form_processor_spec.rb

```diff
@@ -1524,56 +1524,74 @@ def process_record(record:, hud_values:, user:, save: true)
     end
 
     describe 'with custom data elements' do
-      it 'creates a CustomDataElement' do
-        existing_record = i1
-        new_record = Hmis::Hud::Inventory.new(data_source: ds1, user: u1, project: p1)
-
-        cded = create(:hmis_custom_data_element_definition, owner_type: 'Hmis::Hud::Inventory')
+      let!(:cded) { create(:hmis_custom_data_element_definition, owner_type: 'Hmis::Hud::Inventory') }
+      [
+        [:string, 'foo'],
+        [:boolean, false],
+        [:boolean, true],
+        [:integer, 0],
+      ].each do |field_type, value|
+        it "creates a CustomDataElement, on a new or existing record (#{field_type}, #{value})" do
+          existing_record = i1
+          new_record = Hmis::Hud::Inventory.new(data_source: ds1, user: u1, project: p1)
+          cded.update(field_type: field_type)
 
-        [existing_record, new_record].each do |record|
-          hud_values = complete_hud_values.merge(
-            cded.key => 'some value',
-          )
-          process_record(record: record, hud_values: hud_values, user: hmis_user)
+          [existing_record, new_record].each do |record|
+            hud_values = complete_hud_values.merge(cded.key => value)
+            process_record(record: record, hud_values: hud_values, user: hmis_user)
 
-          expect(record.custom_data_elements.size).to eq(1)
-          expect(record.custom_data_elements.first.value_string).to eq('some value')
-          expect(record.custom_data_elements.first.data_element_definition).to eq(cded)
+            expect(record.custom_data_elements.size).to eq(1)
+            expect(record.custom_data_elements.first.value).to eq(value)
+            expect(record.custom_data_elements.first.data_element_definition).to eq(cded)
+          end
         end
       end
 
-      it 'updates a CustomDataElement (repeats: false)' do
-        record = i1
-        cded = create(:hmis_custom_data_element_definition, owner_type: 'Hmis::Hud::Inventory')
-        cde = create(:hmis_custom_data_element, owner: record, value_string: 'old value', data_element_definition: cded)
-        expect(record.custom_data_elements.first!.value_string).to eq(cde.value_string)
+      [
+        [:string, :value_string, 'foo', 'bar'],
+        [:string, :value_string, 'same', 'same'],
+        [:integer, :value_integer, 10, 0],
+        [:integer, :value_integer, 0, 10],
+        [:integer, :value_integer, 0, 0],
+        [:boolean, :value_boolean, true, false],
+        [:boolean, :value_boolean, true, true],
+        [:boolean, :value_boolean, false, false],
+        [:boolean, :value_boolean, false, true],
+      ].each do |field_type, field_name, old_value, new_value|
+        it "updates a CustomDataElement on an existing record (#{field_type}, #{old_value}=>#{new_value}) (repeats: false)" do
```

Comment:
I added a bunch but the specific regression tests are:
```
      updates a CustomDataElement on an existing record (boolean, true=>false) (repeats: false) (FAILED - 1)
      updates a CustomDataElement on an existing record (boolean, false=>false) (repeats: false) (FAILED - 2)
```

These would previously destroy the CDE.